core/src/test/java/com/linecorp/armeria/internal/annotation/AnnotatedHttpServiceResponseConverterTest.java

```diff
@@ -513,4 +552,43 @@ public void charset() {
     public void defaultNullHandling() throws JsonProcessingException {
         assertThat(new ObjectMapper().writeValueAsString(null)).isEqualTo("null");
     }
+
+    @Test
+    public void jsonTextSequences_stream() {
+        testJsonTextSequences("/json-seq/stream");
+    }
+
+    @Test
+    public void jsonTextSequences_publisher() {
+        testJsonTextSequences("/json-seq/publisher");
+    }
+
+    private void testJsonTextSequences(String path) {
+        final BiConsumer<HttpObject, String> ensureExpectedHttpData = (o, expectedString) -> {
+            assertThat(o).isInstanceOf(HttpData.class);
+            final HttpData data = (HttpData) o;
+            try {
+                assertThat(mapper.readValue(data.array(), 1, data.length() - 2, String.class))
+                        .isEqualTo(expectedString);
+            } catch (IOException e) {
+                // Always false.
+                assertThat(e).isNull();
+            }
+        };
+
+        StepVerifier.create(HttpClient.of(rule.uri("/")).get(path))
+                    .assertNext(o -> {
+                        assertThat(o).isInstanceOf(HttpHeaders.class);
+                        final HttpHeaders headers = (HttpHeaders) o;
+                        assertThat(headers.status()).isEqualTo(HttpStatus.OK);
+                        assertThat(headers.contentType()).isEqualTo(MediaType.JSON_SEQ);
+                    })
+                    .assertNext(o -> ensureExpectedHttpData.accept(o, "foo"))
+                    .assertNext(o -> ensureExpectedHttpData.accept(o, "bar"))
+                    .assertNext(o -> ensureExpectedHttpData.accept(o, "baz"))
+                    .assertNext(o -> ensureExpectedHttpData.accept(o, "qux"))
+                    .expectNext(HttpData.EMPTY_DATA)
```

Comment:
Is this expected? Perhaps we could avoid emitting this?

Reply:
Added a comment as follows:
```
.assertNext(o -> {
    // On the server side, HttpResponseSubscriber emits a DATA frame with end of stream
    // flag when the HttpResponseWriter is closed.
    final HttpData lastContent = (HttpData) o;
    assertThat(lastContent.isEmpty()).isTrue();
    assertThat(lastContent.isEndOfStream()).isTrue();
})
```